### PR TITLE
TASK: Improve wording of include/exclude concepts where possible

### PR DESCRIPTION
**Pull request by albe**
Those old wordings have no place and they are also objectively worse than the replacements.

### CODE REVIEW

**Comment by albe** [comment]
Just to be super clear: "white" has no correlation with something being "included"/"allowed" and "black" not with being "excluded"/"disallowed" under any possible point of view.

**Comment by sorenmalling** [comment]
- [ ] ./Neos.Flow/Configuration/Settings.Resource.yaml
- [ ] ./Neos.Flow/Resources/Private/Schema/Settings/Neos.Flow.resource.schema.yaml

I suggest that we rename these aswell, and use the "variable" (can't remember the correct name) concept for backward compatibility, if this should be an issue

**Comment by albe** [comment]
Yep, have those two on my radar, but those would mean a breaking change (unless we only duplicate the options, which is... not really helping). So those should land in master, given most of the team are against breaking for this deliberately. I will create a PR for that once this is approved and merged.

**Comment by robertlemke** [comment]
> And last but not least, it's not as if we have a perfect record of not introducing breaking changes and keeping things perfectly b/c for ages. In fact I would even say that having (well thought) breaking changes is a MUST to really move forward and that in turn is part of the Neos spirit.
Then the only discussion to make is: Is getting rid of those words "moving forward"? And I would heavily argue for that.

I recently had some of this "why-am-I-doing-this"-moments while I was upgrading a Flow application. There were breaking changes, but they were poorly documented and there was no automatic or semi-automatic migration path.

If you provide a Flow core migration which adjusts a user's code automatically to the new wording, a breaking change is much easier to swallow.

What's still unsolved is the problem that, through breaking changes we loose compatibility with many user-contributed packages, which makes timely upgrades to new Neos / Flow versions much harder ([I mentioned that](https://discuss.neos.io/t/psr-15-middlewares-have-landed-in-master/5005/2) in the PSR-15 announcement as well). I think that this can be improved greatly through tooling – if we provide a good concept for package authors how to maintain multiple branches and give them tools to make code adjustments a no-brainer, we can progress faster with more fundamental changes.

But maybe that's a topic for discuss.neos.io …

File: Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php [hunk 1]
```diff
@@ -161,15 +161,26 @@ public function remove($object);
...
     /**
-     * Adds the given object to a whitelist of objects which may be persisted when persistAll() is called with the
-     * $onlyWhitelistedObjects flag. This is the case if "safe" HTTP request methods are used.
+     * Adds the given object to a list of allowed objects which may be persisted when persistAll() is called with the
+     * $onlyAllowedObjects flag. This is the case if "safe" HTTP request methods are used.
      *
      * @param object $object The object
      * @return void
      * @api
+     * @deprecated Use allowObject() instead.
```
[the remark this answers is not in the document]
Comment (robertlemke) [reply Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php:170]: Yes, good idea.

File: Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php [hunk 2]
```diff
@@ -161,15 +161,26 @@ public function remove($object);
...
 
+    /**
+     * Adds the given object to a list of allowed objects which may be persisted when persistAll() is called with the
+     * $onlyAllowedObjects flag. This is the case if "safe" HTTP request methods are used.
+     *
+     * @param object $object The object
+     * @return void
+     * @api
+     */
+    //public function allowObject($object);
```
[the remark this answers is not in the document]
Comment (robertlemke) [reply Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php:182]: Just a thought: Alternatively you can also extend this interface, add the new method (and remove the old one) there already and refactor Flow to use the new interface.

[the remark this answers is not in the document]
Comment (albe) [reply Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php:182]: Wouldn't that be breaking? I mean, most likely no one implemented their own PersistenceManager but the interface methods are @api so we shouldn't do that in patch-level. But we could do it for the LTS already maybe?

File: Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (albe) [reply Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php:170]:
```suggestion
     * @deprecated Use allowObject() instead. See https://github.com/neos/flow-development-collection/pull/2024
```

File: Neos.Eel/Classes/ProtectedContext.php [hunk 3]
```diff
@@ -61,38 +61,63 @@ public function getAndWrap($path = null)
...
      *
-     *   $context->whitelist('*');
+     *   $context->allow('*');
      *
-     *   $context->whitelist(array('String.*', 'Array.reverse'));
+     *   $context->allow(array('String.*', 'Array.reverse'));
      *
      * @param array|string $pathOrMethods
      * @return void
+     * @deprecated Use allow() instead
```
Comment (albe) [inline Neos.Eel/Classes/ProtectedContext.php:88]:
```suggestion
     * @deprecated Use allow() instead. See https://github.com/neos/flow-development-collection/pull/2024
```

File: Neos.Flow/Classes/Persistence/AbstractPersistenceManager.php [hunk 4]
```diff
@@ -91,28 +91,42 @@ public function registerNewObject(Aspect\PersistenceMagicInterface $object)
...
 
     /**
-     * Adds the given object to a whitelist of objects which may be persisted even if the current HTTP request
+     * Adds the given object to a list of allowed objects which may be persisted even if the current HTTP request
      * is considered a "safe" request.
      *
      * @param object $object The object
      * @return void
      * @api
+     * @deprecated Use allowObject() instead
```
Comment (albe) [inline Neos.Flow/Classes/Persistence/AbstractPersistenceManager.php:100]:
```suggestion
     * @deprecated Use allowObject() instead. See https://github.com/neos/flow-development-collection/pull/2024
```

File: Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (kitsunet) [reply Neos.Flow/Classes/Persistence/PersistenceManagerInterface.php:182]: Yeah sounds reasonable! :)

**Pull request merged**